test/unit/test_tpm2_eventlog.c

```diff
@@ -42,15 +45,50 @@ static void test_foreach_digest2(void **state) {
     TCG_DIGEST2* digest = (TCG_DIGEST2*)buf;
 
     will_return(foreach_digest2_test_callback, true);
-    assert_true(foreach_digest2(digest, 1, TCG_DIGEST2_SHA1_SIZE, foreach_digest2_test_callback, NULL));
+
+    tpm2_eventlog_ctx_t ctx = { .digest2_cb = foreach_digest2_test_callback };
+    assert_true(foreach_digest2(&ctx, 0, digest, 1, TCG_DIGEST2_SHA1_SIZE));
 }
 static void test_foreach_digest2_cbnull(void **state){
 
     (void)state;
-    uint8_t buf [TCG_DIGEST2_SHA1_SIZE] = { 0, };
+    uint8_t buf [TCG_DIGEST2_SHA1_SIZE] = { };
```

Comment:
Fix this and all other empty initializers and use `{ 0 }`

Reply:
Oh, right. This is one other those areas that I feel the the C standard committee messed up (see also the trailing `,` in JSON).